pkg/minishift/openshift/version/openshift_versions.go

```diff
@@ -90,28 +76,35 @@ func PrintDownStreamVersions(output io.Writer, minSupportedVersion string) {
 	}
 }
 
-func PrintUpStreamVersions(output io.Writer, minSupportedVersion string) {
-	resp, err := getResponseBody("https://registry.hub.docker.com/v2/repositories/openshift/origin/tags/")
+func PrintUpStreamVersions(output io.Writer, minSupportedVersion string, defaultVersion string) {
+	dockerRegistryUrl := "https://registry.hub.docker.com/v1/repositories/openshift/origin/tags"
+	resp, err := getResponseBody(dockerRegistryUrl)
 	if err != nil {
```

Comment:
This is actually not quite right here. _PrintUpStreamVersions_ should really propagate the error and let the caller handle it. The caller can then decide whether to print nothing, to abort or whatever is appropriate. Right now the method just returns without any indication on whether it worked or not. Well, it prints a message, but this is not really its task. This applies to all errors in this function.